client/src/panels/panel_declarations/results/result_drilldown/result_drilldown.js

```diff
@@ -309,6 +314,42 @@ class SingleSubjExplorer extends React.Component {
                 onChange={(evt) => this.handleQueryChange(evt.target.value)}
                 value={query}
               />
+              <button
+                type="button"
+                className="btn btn-ib-light"
+                style={{
+                  height: "40px",
+                  width: "50%",
+                  margin: "5px 0px",
+                }}
+                onClick={() => {
+                  this.setState({ loading_query: true });
+                  this.expandTimout = setTimeout(() => {
```

Comment:
Please add a comment here and below summarizing why this `setTimeout` is required.

I think this was in the wiki somewhere, but the rule of thumb for comments in the InfoBase is to avoid "what" comments, but to always include "why" comments as necessary. 

When code requires a comments to describe **what** the code is doing, then the code wasn't written clearly enough. Well named variables and functions (especially when you're writing side effect free functional-style code) should "self document". A) it forces us to write clearer, more maintainable code, and B) it avoids the (HUGE\*) problem of documentation comments and code getting out of sync. 

\* trust me, out of date comments are far worse than no comments, the InfoBase massively suffered from this problem back when I started.

The one defensible use of "what" comments are for APIs, but even then I'm personally not a huge fan. Inaccurate API documentation is a bug, and it's ambiguous whether the code or the comments are wrong (not to mention the consumers).

**Why** comments are for cases like this. For the sake of maintainability, we try not to write hacky/unobvious/esoteric code. When there's no avoiding it, usually for equally unobvious/esoteric reasons, and we _don't_ document those reasons, then (best case) we're setting up future developers to duplicate the entire discovery process we went through to determine that this was the only/best option. Worst case, they think past developers were just writing bad code, they reimplement this the straight forward way, and then the original bug regresses (they may not even notice). Basically, if the "why" behind an implementation doesn't boil down to something like "I wrote it this way because it seemed obvious" then it likely needs a comment.

Reply:
... like the `timedOutStateChange` in the query of this component. Vaguely feel like that was originally something I wrote. Don't remember _why_, and I'm wishing past me had document the importance of it. Not clear it's necessary, but I don't want to waste the time right now pulling it apart just to rediscover what it was there for.